models/base/scratch/default/snowplow_ecommerce_base_events_this_run.sql

```diff
@@ -179,128 +175,293 @@ with events_this_run AS (
     {{ snowplow_utils.get_sde_or_context(var('snowplow__atomic_schema', 'atomic'), var('snowplow__context_ecommerce_cart'), lower_limit, upper_limit, 'cart') }},
 {%- endif %}
 
+{% if var('snowplow__enable_mobile_events', false) -%}
+    {{ snowplow_utils.get_sde_or_context(var('snowplow__atomic_schema', 'atomic'), var('snowplow__mobile_session_context'), lower_limit, upper_limit, 'mob_session') }},
+    {{ snowplow_utils.get_sde_or_context(var('snowplow__atomic_schema', 'atomic'), var('snowplow__screen_context'), lower_limit, upper_limit, 'mob_sc_view') }},
+{%- endif %}
+
 {{ snowplow_utils.get_sde_or_context(var('snowplow__atomic_schema', 'atomic'), var('snowplow__sde_ecommerce_action'), lower_limit, upper_limit, 'ecommerce_action', single_entity = false) }},
 
 {{ snowplow_utils.get_sde_or_context(var('snowplow__atomic_schema', 'atomic'), var('snowplow__context_web_page'), lower_limit, upper_limit, 'page_view') }}
 
+select
+    *,
+    dense_rank() over (partition by domain_sessionid order by derived_tstamp) AS event_in_session_index
+from (
```

Comment:
The subquery here and removing the filter on the collector_tstamp earlier might cause performance issues. I'm tempted to suggest to keep a separate sql for the two situations until we find a less hectic solution.

Reply:
Yeah I did try to keep it optimised, but also I looked back at it this morning and don't know why I did it in the first place, new version pushed.

Reply:
Much more elegant, nice!